tests/page-objects/default-mobile/search/search.wdio.page.js

```diff
@@ -1,14 +1,27 @@
 const searchPageDefault = require('@page-objects/default/search/search.wdio.page');
 
 const openSearchBox = () => $('.mm-search-bar-container .search-bar-left-icon .fa-search');
+const barcodeSearchInput = () => $('.barcode-scanner-input');
 
 const performSearch = async (term) => {
   await (await openSearchBox()).waitForClickable();
   await (await openSearchBox()).click();
   await searchPageDefault.performSearch(term);
 };
 
+const performBarcodeSearch = async (barcodeImagePath) => {
+  /*In this case the upload file button is hidden,
+  then we need to manipulate the DOM of the respective webelement to make it interactable.*/
+  browser.execute(function () {
+    document.getElementsByClassName('barcode-scanner-input')[0].style.display = 'block';
```

Comment:
Not a fain of this either, because it will definitely obscure if the actual button that makes the input selection no longer works, or is not displayed. So the test only validates the functionality partially. 
Is there no way to use the elements that are displayed on the page to upload the image?

Reply:
If you click on the barcode icon, the OS upload file dialog appears, and Webdriverio has no control over it. I'm investigating how to setValue to a web element that is not interactable without modifying the DOM.

Reply:
Yea, I went through that search as well and in the end I couldn't find anything. The idea is that wdio really wants to simulate as close to user interaction as possible. 
Maybe we should just leave it like this.